src/wui/interactive_player.cc

```diff
@@ -420,18 +428,22 @@ void InteractivePlayer::draw_map_view(MapView* given_map_view, RenderTarget* dst
 	assert(!get_sel_triangles());
 
 	const Widelands::Player& plr = player();
-	const auto& gbase = egbase();
+	const Widelands::EditorGameBase& gbase = egbase();
 	const Widelands::Map& map = gbase.map();
 	const uint32_t gametime = gbase.get_gametime();
 
 	Workareas workareas = get_workarea_overlays(map);
-	auto* fields_to_draw = given_map_view->draw_terrain(gbase, &plr, workareas, false, dst);
+	FieldsToDraw* fields_to_draw = given_map_view->draw_terrain(gbase, &plr, workareas, false, dst);
 	const auto& road_building_s = road_building_steepness_overlays();
 
 	const float scale = 1.f / given_map_view->view().zoom;
 
+	// Store the coords of partially visible buildings
+	// so we can draw them later when we get to their main position.
+	std::vector<Widelands::Coords> deferred_coords;
```

Comment:
Since the order does not matter, an `std::set` would make for better readability IMHO because you can then use `count()` rather than that ugly `std::find()`